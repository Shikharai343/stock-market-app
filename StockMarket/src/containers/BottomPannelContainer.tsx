import React from 'react';
import axios from 'axios';
import {Dimensions, StyleSheet, Text, View} from 'react-native';
import {BarChart} from 'react-native-chart-kit';

interface IState {
    stockDate: string[];
    stockPrice: string[];
}

class BottomPannelContainer extends React.Component <any, IState>{
    constructor(props: any) {
        super(props);
        this.state = {
            stockDate: [],
            stockPrice: []
        }
    }
    componentDidMount(){
        axios.get('https://api.airtable.com/v0/appaf01a2JbZoZNee/stock%20market?api_key=keyiMCbcYlCf5VXsP')
            .then((res) => {
                this.setState({
                    stockDate: res.data.records.map((input: any) => {
                        return input.fields.Date;
                    }),
                    stockPrice: res.data.records.map((input: any) => {
                        return input.fields.Close;
                    })
                });

            });
    }

    render () {
        const data = {
            labels: this.state.stockDate,
            datasets: [{
                data: this.state.stockPrice,
            }]
        }

        const screenWidth = Dimensions.get('window').width;
        const chartConfig = {
            backgroundGradientFrom: '#1E2923',
            backgroundGradientTo: '#08130D',
            color: (opacity = 1) => `rgba(26, 255, 146, ${opacity})`,
            strokeWidth: 2 // optional, default 3
        }

        return (
            <View style={styles.container}>
                <Text style={styles.heading}>Stock Price Graph</Text>
                <BarChart
                    data={data}
                    width={screenWidth}
                    height={300}
                    yAxisLabel={'$'}
                    chartConfig={chartConfig}
                />
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        marginTop: '5%'
    },
    heading: {
        textAlign: 'center',
        fontSize: 30,
        marginBottom: '5%',
    }
})

export default BottomPannelContainer;
