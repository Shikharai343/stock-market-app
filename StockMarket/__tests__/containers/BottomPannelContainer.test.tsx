import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, ShallowWrapper} from 'enzyme';
import BottomPannelContainer from '../../src/containers/BottomPannelContainer';

let wrapper: ShallowWrapper;

describe('BottomPannelContainer testing', () => {
    wrapper = shallow(<BottomPannelContainer/>);

    it('should render snapshot', () => {
        const tree = renderer.create(<BottomPannelContainer/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
