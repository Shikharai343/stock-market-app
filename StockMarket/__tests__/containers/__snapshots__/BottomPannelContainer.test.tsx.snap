// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomPannelContainer testing should render snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": "5%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 30,
        "marginBottom": "5%",
        "textAlign": "center",
      }
    }
  >
    Stock Price Graph
  </Text>
  <View
    style={Object {}}
  >
    <RNSVGSvgView
      bbHeight={300}
      bbWidth={750}
      height={300}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 300,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4278190080,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={Array []}
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -14800605,
                1,
                -16248051,
              ]
            }
            gradientTransform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            gradientUnits={0}
            name="backgroundGradient"
            x1="0"
            x2={750}
            y1={300}
            y2={0}
          />
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                437976978,
                1,
                1769362,
              ]
            }
            gradientTransform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            gradientUnits={0}
            name="fillShadowGradient"
            x1={0}
            x2={0}
            y1={0}
            y2={300}
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "backgroundGradient",
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={300}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={0}
        />
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGLine
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                857407378,
              ]
            }
            strokeDasharray={
              Array [
                "5",
                "10",
              ]
            }
            strokeDashoffset={0}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            x1={0}
            x2={750}
            y={0}
            y1={16}
            y2={16}
          />
          <RNSVGLine
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                857407378,
              ]
            }
            strokeDasharray={
              Array [
                "5",
                "10",
              ]
            }
            strokeDashoffset={0}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            x1={0}
            x2={750}
            y={0}
            y1={91}
            y2={91}
          />
          <RNSVGLine
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                857407378,
              ]
            }
            strokeDasharray={
              Array [
                "5",
                "10",
              ]
            }
            strokeDashoffset={0}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            x1={0}
            x2={750}
            y={0}
            y1={166}
            y2={166}
          />
          <RNSVGLine
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                857407378,
              ]
            }
            strokeDasharray={
              Array [
                "5",
                "10",
              ]
            }
            strokeDashoffset={0}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            x1={0}
            x2={750}
            y={0}
            y1={241}
            y2={241}
          />
        </RNSVGGroup>
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        >
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={
              Array [
                0,
                2149253010,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={
              Object {
                "fontSize": 12,
                "textAnchor": "end",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={
              Array [
                52,
              ]
            }
            y={
              Array [
                237,
              ]
            }
          >
            <RNSVGTSpan
              content="$NaN"
              dx={Array []}
              dy={Array []}
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={
              Array [
                0,
                2149253010,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={
              Object {
                "fontSize": 12,
                "textAnchor": "end",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={
              Array [
                52,
              ]
            }
            y={
              Array [
                166,
              ]
            }
          >
            <RNSVGTSpan
              content="$NaN"
              dx={Array []}
              dy={Array []}
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={
              Array [
                0,
                2149253010,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={
              Object {
                "fontSize": 12,
                "textAnchor": "end",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={
              Array [
                52,
              ]
            }
            y={
              Array [
                95,
              ]
            }
          >
            <RNSVGTSpan
              content="$NaN"
              dx={Array []}
              dy={Array []}
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={
              Array [
                0,
                2149253010,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={
              Object {
                "fontSize": 12,
                "textAnchor": "end",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={
              Array [
                52,
              ]
            }
            y={
              Array [
                24,
              ]
            }
          >
            <RNSVGTSpan
              content="$NaN"
              dx={Array []}
              dy={Array []}
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotate={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
        <RNSVGGroup
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
